scripts/dockerfiles/Dockerfile.init

```diff
@@ -2,7 +2,7 @@ ARG golang_image
 
 FROM $golang_image as builder
 WORKDIR /go/src/github.com/aws/amazon-vpc-cni-k8s
-ARG GOARCH
+ARG TARGETARCH
```

Comment:
From my Makefile comment above, I think we can simplify things if we add `ENV ARCH=$TARGETARCH` here too (and clean up the Makefile a bit)

Reply:
As noted in the other comment, TARGETARCH is available only when buildkit is enabled, which isn't the default yet in linux. Thus, this platform arg is set by docker only in case of multi-platform builds (where buildx and buildkit are used).

Reply:
Now, TARGETARCH value is used in the makefile to set the value of goarch